examples/pm_custom/main.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Example application for custom power management
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+
+void pm_set_lowest(void)
+{
+    static uint32_t n = 0;
+    if (n++ == 0xFFFFFF) {
+        puts("sleeping");
+        n = 0;
```

Comment:
Maybe the example could use pm_set to better show what a usable pm_set_lowest might do.

Reply:
Here's the catch for overriding pm_layered: there is no platform independent definition of any mode. Suggestions welcome.

Zephyr defines four fixed modes: one where the cpu is set to idle, one where RAM and periph state is kept  but only interrupts can wake up, one where no state is kept but e.g., RTC can still wake up, and off. Maybe the first users of pm_custom can come up with something here.

Or we go with ```#ifdef```'ed special cases for different platforms, just to show the feature.